### PR TITLE
[IA-3320] create individual http client

### PR DESCRIPTION
**Pull request by Qi77Qi**
Today we're sharing one http client for all DAOs. It has been fine, but we don't have to use the same one. This PR makes it so that each DAO has its own http client, and depends on the DAO, we'll generate metrics for each of them. Another benefit is that if Sam is down, and it takes up all the connection pool, Leo will still be able to make requests to other services. Having metrics for each DAO will also help us identify more easily where things might be wrong.

I added http4s client metrics for the shared client recently. Here's what it'll look like

Screen Shot 2022-04-11 at 10 15 11 AM

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by xuscrbw — APPROVED** [review]
lgtm

**Review by jdcanas — APPROVED** [review]

**Comment by Qi77Qi** [comment]
jenkins retest

File: .github/workflows/unit_test.yml [hunk 1]
```diff
@@ -43,10 +44,9 @@ jobs:
 
     - name: Run tests
       env:
-        JAVA_OPTS: -Xmx4G -Xms4G -Xss2M
+        JAVA_OPTS: -Xmx4G -Xms4G -Xss2M -Duser.timezone=UTC -Denv.type=test -Dmysql.host=localhost -Dmysql.port=3307
```
Comment (jdcanas) [inline .github/workflows/unit_test.yml:47]: Hmm... all the test failures have to do with the created date being off from the expected value it seems. Could it be this time zone flag?

Comment (Qi77Qi) [reply .github/workflows/unit_test.yml:47]: it's not timezone...for some reason, one timestamp is more precise, see 
```
org.scalatest.exceptions.TestFailedException: Some(DataprocInstance(DataprocInstanceKey(dsp-leo-test,ZoneName(my-zone),InstanceName(master-instance)),12345,Running,Some(IP(1.2.3.4)),Master,2022-04-12T16:12:06.553181Z)) did not equal Some(DataprocInstance(DataprocInstanceKey(dsp-leo-test,ZoneName(my-zone),InstanceName(master-instance)),12345,Running,Some(IP(1.2.3.4)),Master,2022-04-12T16:12:06.553180524Z))
```

**Pull request merged**